### PR TITLE
[INLONG-11352][SDK] Add dirty data collection sdk

### LINKED ISSUE #11352: [Improve][SDK] Add dirty data collection sdk
### Description

Add dirty data collection sdk

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
Fixes #11352 

### Motivation

The ability to archive dirty data is required for each module and needs to be made into an independent SDK

### Modifications

 Add dirty data collection sdk

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

No doc needed

### CODE REVIEW

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/Constants.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.dirtydata;
+
+/**
+ * connector base option constant
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/Constants.java:21]:
```suggestion
 * Connector base option constant
```

File: inlong-sdk/dirty-data-sdk/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    <version>${plugin.deploy.version}</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-site-plugin</artifactId>
+                    <version>3.7.1</version>
+                </plugin>
+                <plugin>
+                    <artifactId>maven-project-info-reports-plugin</artifactId>
+                    <version>3.0.0</version>
+                </plugin>
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/pom.xml:91]: Do all these plugins are needed?

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/sink/LabelUtils.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+package org.apache.inlong.sdk.dirtydata.sink;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.apache.inlong.sdk.dirtydata.Constants.DELIMITER;
+import static org.apache.inlong.sdk.dirtydata.Constants.KEY_VALUE_DELIMITER;
+
+/**
+ * The lable utils class, it is used for parse the lables to a label map
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/sink/LabelUtils.java:27]:
```suggestion
 * The label utils class, it is used for parse the lables to a label map
```

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/SchemaUpdateExceptionPolicy.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * Multiple sink scenes will meet different table data.
+ * Maybe one table data have different schema, once it's schema mismatch with catalog schema, how to handle
+ * this table data. For example schema mismatch:
+ *
+ * <pre>
+ * data : {a : int, b : string, c : date}
+ * catalog : {a : string, b : timestamp}
+ * </pre>
+ */
+public enum SchemaUpdateExceptionPolicy {
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/SchemaUpdateExceptionPolicy.java:30]: Where is `SchemaUpdateExceptionPolicy` used?

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/PatternReplaceUtils.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * The pattern replace utils
+ */
+public final class PatternReplaceUtils {
+
+    private static final Pattern REGEX_PATTERN = Pattern.compile("\\$\\{\\s*([\\w.-]+)\\s*}",
+            Pattern.CASE_INSENSITIVE);
+
+    public static String replace(String pattern, Map<String, String> params) {
+        if (pattern == null) {
+            return pattern;
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/PatternReplaceUtils.java:34]: `pattern is always null`

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/sink/InlongSdkDirtySink.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+import java.util.Map;
+import java.util.StringJoiner;
+
+import static org.apache.inlong.sdk.dirtydata.Constants.DIRTY_SIDE_OUTPUT_IGNORE_ERRORS;
+
+@Slf4j
+public class InlongSdkDirtySink implements DirtySink {
+
+    // The inlong manager addr to init inlong sdk
+    private static final String DIRTY_SIDE_OUTPUT_INLONG_MANAGER = "dirty.side-output.inlong-sdk.inlong-manager-addr";
```
Comment (aloyszhang) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/sink/InlongSdkDirtySink.java:43]:
```suggestion
    private static final String DIRTY_SIDE_OUTPUT_INLONG_MANAGER_ADDR = "dirty.side-output.inlong-sdk.inlong-manager-addr";
```

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/Constants.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+package org.apache.inlong.sdk.dirtydata;
+
+/**
+ * Connector base option constant
+ */
+public final class Constants {
+
+    // The action to deal with schema update in multiple sink.
+    public static final String SINK_MULTIPLE_SCHEMA_UPDATE_POLICY = "sink.multiple.schema-update.policy";
```
Comment (vernedeng) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/Constants.java:26]: Are all these constants needed?

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/DirtyData.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+     * The real dirty data
+     */
+    private final byte[] data;
+
+    public DirtyData(byte[] data, String identifier, String labels,
+            String logTag, String dirtyType, String dirtyMessage) {
+        this.data = data;
+        this.dirtyType = dirtyType;
+        this.dirtyMessage = dirtyMessage;
+        Map<String, String> paramMap = genParamMap();
```
Comment (vernedeng) [inline inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/DirtyData.java:72]: it seems that paramMap is a useless param

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/SchemaUpdateExceptionPolicy.java (hunk 4 above)
Comment (justinwwhuang) [reply inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/SchemaUpdateExceptionPolicy.java]: > Where is `SchemaUpdateExceptionPolicy` used?

Deleted

File: inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/DirtyData.java (hunk 8 above)
Comment (justinwwhuang) [reply inlong-sdk/dirty-data-sdk/src/main/java/org/apache/inlong/sdk/dirtydata/DirtyData.java:72]: > it seems that paramMap is a useless param

The following code is useful

File: inlong-sdk/dirty-data-sdk/pom.xml (hunk 2 above)
Comment (justinwwhuang) [reply inlong-sdk/dirty-data-sdk/pom.xml:91]: > Do all these plugins are needed?

Yes, it is necessary to use when packaging the SDK

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — DISMISSED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by vernedeng — DISMISSED** [review]

File: inlong-sdk/dirty-data-sdk/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
+## Overview
+
+This SDK is used to collect dirty data and store it in a designated storage location.
+
+## Features
+
+### Independent SDK
+
+Independent SDK, not dependent on platform specific libraries (such as Flink), can be used by Agent,
+Data Proxy, Sort modules.
```
Comment (dockerzhang) [inline inlong-sdk/dirty-data-sdk/README.md:10]:
```suggestion
DataProxy, Sort modules.
```

**Review by dockerzhang — APPROVED** [review]

**Review by doleyzi — APPROVED** [review]

**Pull request merged**